### ISSUE TITLE
incomplete PROJ.4 from NetCDF 

### ISSUE THREAD

**Issue by mdsumner** [rspatial/raster#14]
I've seen this a few times, a message like

```
Warning messages:
...
2: In .getCRSfromGridMap4(atts) : cannot process these parts of the CRS:
```

These NetCDF files https://water.weather.gov/precip/ present the warning,  I've attached an example, it has 3 separate regions  in three files conus, ak, pr all on the same CRS. 

The CRS should be 

```
"+proj=stere +lat_0=90 +lat_ts=60 +lon_0=-105 +x_0=0 +y_0=0 +a=6367470 +b=6367470 +units=m +no_defs"
```

and that is actually in the NetCDF metadata, though I'm sure other examples only have the WKT string. 

I don't know the fix exactly, but getting the proj4 if present would be easiest - a more general check of the WKT probably can't be  easily done without GDAL. 

raster is getting the incorrect "+proj=stere +lon_0=-105 +x_0=0 +y_0=0 +lat_0=90 +ellps=WGS84"  (no latitude of true scale, wrong on WGS84, no +a/+b). 

[precip.zip](https://github.com/rspatial/raster/files/2506288/precip.zip)

**Comment by rhijmans** [comment]
I fixed the raster function such that it now picks up the "earth_radius"
variable (and translates that to  "+a" and "+b"; and thus avoiding the
default WGS84).
The other information (proj4, wkt) is not part of the ncdf CF conventions.
They came up with their own rules that only deal with a few cases. This is
very unfortunate to say the least. I changed to code so that it will use
proj4 strings if it finds a variable called "proj4". And 'raster' will also
use that same variable name when writing ncdf4 --- perhaps it can become a
de facto standard.
